ipatests/test_xmlrpc/test_sysaccount_plugin.py

```diff
@@ -0,0 +1,308 @@
+# Copyright (C) 2025  Red Hat
+# see file 'COPYING' for use and warranty information
+
+"""
+Test the `ipaserver/plugins/sysaccounts.py` module.
+"""
+
+from __future__ import print_function
+
+import pytest
+
+from ipalib import api, errors
+from ipatests.test_xmlrpc.xmlrpc_test import XMLRPC_test
+from ipatests.util import change_principal, unlock_principal_password
+from ipatests.test_xmlrpc.xmlrpc_test import assert_attr_equal
+
+NORMAL_USER = u'test-normal-user'
+NORMAL_USER_PASSWORD = u'Secret123'
+SECURITY_ARCHITECT_USER = u'test-security-architect'
+SECURITY_ARCHITECT_PASSWORD = u'Secret123'
+SECURITY_ARCHITECT_ROLE = u'Security Architect'
+
+TEST_SYSACCOUNT_1 = u'test-sysaccount-1'
+TEST_SYSACCOUNT_2 = u'test-sysaccount-2'
+TEST_SYSACCOUNT_3 = u'test-sysaccount-3'
+
+
+@pytest.mark.tier1
+class TestSysaccountPermissions(XMLRPC_test):
+    """
+    Test system account permissions with different user roles.
+    """
+
+    def test_setup(self):
+        api.Command.user_add(
+            NORMAL_USER,
+            givenname=u'Normal',
+            sn=u'User',
+            userpassword=NORMAL_USER_PASSWORD
+        )
+        unlock_principal_password(NORMAL_USER, NORMAL_USER_PASSWORD,
+                                  NORMAL_USER_PASSWORD)
+
+        # Create Security Architect user
+        api.Command.user_add(
+            SECURITY_ARCHITECT_USER,
+            givenname=u'Security',
+            sn=u'Architect',
+            userpassword=SECURITY_ARCHITECT_PASSWORD
+        )
+        unlock_principal_password(SECURITY_ARCHITECT_USER,
+                                  SECURITY_ARCHITECT_PASSWORD,
+                                  SECURITY_ARCHITECT_PASSWORD)
+
+        # Add Security Architect user to Security Architect role
+        api.Command.role_add_member(SECURITY_ARCHITECT_ROLE,
+                                    user=SECURITY_ARCHITECT_USER)
+
+    def test_normal_user_cannot_add_sysaccount(self):
+        """Test that a normal user cannot add a system account"""
+        with change_principal(NORMAL_USER, NORMAL_USER_PASSWORD):
+            with pytest.raises(errors.ACIError):
+                api.Command.sysaccount_add(TEST_SYSACCOUNT_1, random=True)
+
+    def test_normal_user_cannot_modify_sysaccount(self):
+        """Test that a normal user cannot modify a system account"""
+        # First create the sysaccount as admin
+        api.Command.sysaccount_add(TEST_SYSACCOUNT_1, random=True)
+
+        try:
+            with change_principal(NORMAL_USER, NORMAL_USER_PASSWORD):
+                with pytest.raises(errors.ACIError):
+                    api.Command.sysaccount_mod(
+                        TEST_SYSACCOUNT_1,
+                        description=u'Modified description'
+                    )
+        finally:
```

Comment:
Instead of doing add + delete of the TEST_SYSACCOUNT_1, the test can use fixtures. You need to define a SysaccountTracker